### PR TITLE
[luci] Introduce FuseBatchNormWithConvPass

### PR DESCRIPTION
**Pull request by llFreetimell**
Parent Issue : #5618

This commit will introduce `FuseBatchNormWithConvPass`.

ONE-DCO-1.0-Signed-off-by: Seok NamKoong <seok9311@naver.com>

### CODE REVIEW

File: compiler/luci/pass/src/FuseBatchNormWithConvPass.cpp [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+    conv = mul_rhs;
+    scale = dynamic_cast<luci::CircleConst *>(mul->x());
+  }
+
+  // If CircleConv2D is not found or constant operand of CircleMul is not found,
+  // this pass cannot be applied.
+  if (conv == nullptr || scale == nullptr)
+    return false;
+
+  // To apply this pass, shape of shift should be [1, 1, 1, out_channel].
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseBatchNormWithConvPass.cpp:86]: maybe `shift` -> `scale` ?

**Review by seanshpark — DISMISSED** [review]
LGTM

File: compiler/luci/pass/src/FuseBatchNormWithConvPass.cpp [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+  }
+
+  // Copy bias
+  fused_bias->dtype(bias->dtype());
+  fused_bias->size<loco::DataType::FLOAT32>(filter->size<loco::DataType::FLOAT32>());
+  fused_bias->rank(4);
+  fused_bias->dim(0).set(1);
+  fused_bias->dim(1).set(1);
+  fused_bias->dim(2).set(1);
+  fused_bias->dim(3).set(filter_out_channel);
```
Comment (jinevening) [inline compiler/luci/pass/src/FuseBatchNormWithConvPass.cpp:163]: Can you make bias have rank 1 (or at least have the same shape with the original bias)?

```suggestion
  fused_bias->rank(1);
  fused_bias->dim(0).set(filter_out_channel);
```

Comment (llFreetimell) [reply compiler/luci/pass/src/FuseBatchNormWithConvPass.cpp:163]: Oops, it's my mistake :(

**Review by mhs4670go — APPROVED** [review]
LGTM

File: compiler/luci/pass/src/FuseBatchNormWithConvPass.cpp [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        }
+      }
+    }
+  }
+
+  // Copy bias
+  fused_bias->dtype(bias->dtype());
+  fused_bias->size<loco::DataType::FLOAT32>(filter->size<loco::DataType::FLOAT32>());
+  fused_bias->rank(1);
+  fused_bias->dim(0).set(filter_out_channel);
```
Comment (seanshpark) [inline compiler/luci/pass/src/FuseBatchNormWithConvPass.cpp:160]: this assumes `bias` is rank 1 with dim size `filter_out_channel` but there is no condition check. is this OK?

Comment (llFreetimell) [reply compiler/luci/pass/src/FuseBatchNormWithConvPass.cpp:160]: AFAIK, that condition is always met, but I will add assert for it :)

**Review by jinevening — APPROVED** [review]
LGTM Thank you

**Review by mhs4670go — APPROVED** [review]
LGTM

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Pull request merged**